app/controllers/comparisons/heating_vs_hot_water_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Comparisons
+  class HeatingVsHotWaterController < BaseController
+    private
+
+    def header_groups
+      [{ label: '',
+         headers: [t('analytics.benchmarking.configuration.column_headings.school')] },
+       { label: t('analytics.benchmarking.configuration.column_groups.kwh'),
+         headers: [
+           t('analytics.benchmarking.configuration.column_headings.gas'),
+           t('analytics.benchmarking.configuration.column_headings.hot_water_gas'),
+           t('analytics.benchmarking.configuration.column_headings.heating_gas')
+         ] },
+       { label: '',
+         headers: [t('analytics.benchmarking.configuration.column_headings.percentage_of_gas_use_for_hot_water')] }]
+    end
+
+    def key
+      :heating_vs_hot_water
+    end
+
+    def advice_page_key
+      :hot_water
+    end
+
+    def load_data
+      Comparison::HeatingVsHotWater.for_schools(@schools).where.not(last_year_gas_kwh: nil).order(last_year_gas_kwh: :desc)
```

Comment:
The where clause here excludes schools with no gas usage, which is correct but it means that the report will include schools that have no hot water usage either. I think it needs to check whether we have both. This ensures that we:

- rule out schools where we don't have gas data (no either analysis is present)
- rule out schools where we aren't successfully able to run the hot water analysis, e.g. due to limited data. The AlertHotWaterEfficiency analysis only runs if we have data covering a summer holiday with a few weeks either side

As noted on Trello the report should also exclude schools with swimming pools, like the Hot Water Efficiency report. This model defines an extra scope which is used in the controller:

```
scope :without_swimming_pool, -> { joins(:school).where({ school: { has_swimming_pool: false } }) }
```

That report also:

- uses a different message for reporting which schools are unlisted to add a note about swimming pools, see `unlisted_message` in controller
- updates the list of unlisted schools to indicate which ones have swimming pools. See `hot_water_efficiency/_unlisted.html.erb`

The Trello card also notes that the report should be sorted by % use of gas for heating, in descending order. Should be possible to calculate that in the SQL query as we do with other % comparisons.

Reply:
made changes for this